pkg/distributor/distributor_test.go

```diff
@@ -288,8 +288,18 @@ func TestDistributor_Push(t *testing.T) {
 
 			request := makeWriteRequest(tc.samples.startTimestampMs, tc.samples.num, tc.metadata, false)
 			response, err := ds[0].Push(ctx, request)
-			assert.Equal(t, tc.expectedResponse, response)
-			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				assert.Equal(t, emptyResponse, response)
+				assert.Nil(t, err)
+			} else {
+				assert.Nil(t, response)
+				assert.EqualError(t, err, tc.expectedError.Error())
+
+				// Assert that downstream GRPC statuses are passed back upstream
+				_, ok := httpgrpc.HTTPResponseFromError(err)
+				assert.True(t, ok)
```

Comment:
```suggestion
				_, ok := httpgrpc.HTTPResponseFromError(err)
				assert.True(t, ok, "expected error to be an httpgrpc error")
```

Should we add a message to this line, so the failure output is more descriptive than `should be true`?